[WIP] Squash doxygen errors.

sniyaz opened the pull request:
See https://github.com/personalrobotics/aikido/issues/329.

This mainly involved fixing some doc-strings and renaming parameters. 

**TODO:** I had some trouble squashing the warnings involving `SplineTrajectory-impl.hpp.` @jslee02 and other people who know doxygen better, could take a look? There doesn't appear to be an documentation in that file at all, so I'm confused 😖 

***

**Before creating a pull request**

- [x] Document new methods and classes
- [x] Format code with `make format`

**Before merging a pull request**

- [x] Set version target by selecting a milestone on the right side
- [x] Summarize this change in `CHANGELOG.md`
- [x] Add unit test(s) for this change

jslee02 commented on include/aikido/constraint/dart/CollisionFree.hpp:
```diff
@@ -60,27 +60,26 @@ class CollisionFree : public Testable
-  /// \param group Collision group.
-  void addSelfCheck(std::shared_ptr<::dart::collision::CollisionGroup> _group);
+  /// \param _group Collision group.
+  void addSelfCheck(std::shared_ptr<dart::collision::CollisionGroup> _group);
```
Nit: Was this one of the reasons of the Doxygen warnings? We used global namespace resolutions for dart (i.e., `::dart`) to avoid conflicts with `::aikido::constraint::dart`. It might be good to leave it as it was if it's not a cause of the warnings.

jslee02 commented on include/aikido/constraint/dart/CollisionFree.hpp:
```diff
@@ -60,27 +60,26 @@ class CollisionFree : public Testable
-  void removeSelfCheck(
-      std::shared_ptr<::dart::collision::CollisionGroup> _group);
+  /// \param _group Collision group.
+  void removeSelfCheck(std::shared_ptr<dart::collision::CollisionGroup> _group);
```
Nit: Was this one of the reasons of the Doxygen warnings? We used global namespace resolutions for dart (i.e., `::dart`) to avoid conflicts with `::aikido::constraint::dart`. It might be good to leave it as it was if it's not a cause of the warnings.

sniyaz commented:
Hang on, flipped something. Will fix to satisfy Travis.

jslee02 commented on include/aikido/statespace/detail/SO3-impl.hpp:
```diff
@@ -31,7 +31,6 @@ class SO3StateHandle : public statespace::StateHandle<SO3, _QualifiedState>
 
   /// Constructs a point in SO(3) from a unit quaternion.
   ///
```
Nit: Could you remove this empty line as well?

sniyaz commented:
I've squashed some errors from the Robot API and deleted `SplineTrajectory-impl.hpp`. Should be good to go now. Please rubber stamp when you have the chance 😄

brianhou commented on include/aikido/control/BarrettFingerKinematicSimulationPositionCommandExecutor.hpp:
```diff
@@ -82,8 +82,10 @@ class BarrettFingerKinematicSimulationPositionCommandExecutor
   /// is reached, or collision is detected.
   void step(const std::chrono::system_clock::time_point& timepoint) override;
 
-  /// \copydoc
```
Let's figure out how to fix this so we don't need to duplicate the `\copydoc`s.

jslee02 commented on include/aikido/control/BarrettFingerKinematicSimulationPositionCommandExecutor.hpp:
```diff
@@ -82,10 +82,9 @@ class BarrettFingerKinematicSimulationPositionCommandExecutor
-  ///
-  /// \param collideWith CollisionGroup to check finger collisions
-  /// \return false if collideWith cannot be set (during execution)
+  /* clang-format off */
```
Nit: Could we use `// clang-format off` (C++ style) instead of `/* clang-format off */` (C style)?

sniyaz commented:
@brianhou I tried to look up ways to turn break the `copydoc` into multiple lines, but the only solution seems to manually turning clang format on/off 😢
